src/session/session_dhandle.c

```diff
@@ -506,8 +506,8 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
      * Test for the internal checkpoint name (WiredTigerCheckpoint). Note: must_resolve is true in a
      * subset of the cases where is_unnamed_ckpt is true.
      */
-    must_resolve = WT_STRING_MATCH(WT_CHECKPOINT, cval.str, cval.len);
     is_unnamed_ckpt = cval.len >= strlen(WT_CHECKPOINT) && WT_PREFIX_MATCH(cval.str, WT_CHECKPOINT);
+    must_resolve = is_unnamed_ckpt && WT_CONFIG_LIT_MATCH(WT_CHECKPOINT, cval);
```

Comment:
I couldn't help making a tiny optimization when changing this code.  I think it's equivalent, but I'm happy to make it more like the original if it makes anyone nervous.

Reply:
Given that MongoDB doesn't use named checkpoints, is there any optimization if `is_unnamed_ckpt` always true? I am otherwise okay with the change as it is equivalent AFAICT.

Reply:
The problem with this is someone reading it won't know the equivalence. If you do this you should add a comment saying: /* Equivalent to must_resolve = WT_STRING_MATCH(WT_CHECKPOINT, cval.str, cval.len); but faster. */

Also, I think line 510 is saying not only is WT_CHECKPOINT a prefix of cval.str, it is the same. In that case line 510 can be replaced with.

must_resolve = is_unnamed_ckptr && cval.str[strlen(WT_CHECKPOINT)] == '\0';
If the compiler replaces strlen() with the length of WT_CHECKPOINT at compile time this does not call any function.

Reply:
I removed this optimization, it's not perf-critical code anway.